pkg/providers/tinkerbell/validator.go

```diff
@@ -180,6 +181,23 @@ func (v *Validator) ValidateBMCSecretCreds(ctx context.Context, hc hardware.Hard
 	return nil
 }
 
+func (v *Validator) ValidateTinkerbellTemplate(ctx context.Context, tinkerbellIp string, templateConfig *v1alpha1.TinkerbellTemplateConfig) error {
+	for _, task := range templateConfig.Spec.Template.Tasks {
+		for _, action := range task.Actions {
+			// check if the metadata_url provided by the user is valid or not
+			// TODO(pjshah):: add more validations around metadata_url field and parse
+			if action.Name == "add-tink-cloud-init-config" {
+				metadataUrl := fmt.Sprintf("http://%s:50061", tinkerbellIp)
+				if !strings.Contains(action.Environment["CONTENTS"], metadataUrl) {
+					return fmt.Errorf("failed validating metadata_url, template metadata_url should be in http://<TINKERBELL_IP>:50061 format")
```

Comment:
Since you are checking against the exact metadata url, you should just tell the users that ".... template metadata_url should be set to ...."